central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -3,32 +3,65 @@ package compliancemanager
 import (
 	"context"
 
+	"github.com/adhocore/gronx"
+	"github.com/gogo/protobuf/types"
 	"github.com/pkg/errors"
-	integrationDS "github.com/stackrox/rox/central/complianceoperator/v2/integration/datastore"
+	compIntegration "github.com/stackrox/rox/central/complianceoperator/v2/integration/datastore"
+	compScanSetting "github.com/stackrox/rox/central/complianceoperator/v2/scanconfigurations/datastore"
 	"github.com/stackrox/rox/central/sensor/service/connection"
+	"github.com/stackrox/rox/generated/internalapi/central"
 	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/protoconv/schedule"
 	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/uuid"
 )
 
+const (
+	// Daily at midnight
+	defaultScanSchedule = "0 0 * * *"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+type clusterScan struct {
+	clusterID string
+	scanID    string
+}
+
 type managerImpl struct {
 	sensorConnMgr connection.Manager
-	integrationDS integrationDS.DataStore
+	integrationDS compIntegration.DataStore
+	scanSettingDS compScanSetting.DataStore
+	requests      map[string]clusterScan
```

Comment:
Is the key `requestID` which has 1:n mapping to scans?
Does this hold currently running requests or just a cache without invariants?

Reply:
@md2119 
This only holds currently running requests.

I started down that path of using one requestID for all clusters but dealing with the 1:n seemed unnecessarily complicated.  Down below I generate a unique id to pass into each sensor message.  Obviously I should probably name that variable better.  The idea is this will hold the requests that are out to the sensors so that we can easily update the status when the message comes back.  

```
	for _, cluster := range clusters {
		// id for the request message to sensor
		id := uuid.NewV4().String()
```
Now that I'm explaining this, a lot of it is a remnant when I had the status buried in the configuration object and was trying to be surgical and minimize updates to the configuration object.  Since I split out the cluster status as a separate table for state, I could probably use the table instead as it is per cluster with a FK back to the configuration.

Opinions?

Reply:
Thanks for the explanation. A couple of comments to remove the confusion:
- The intent can be clarified by renaming the field to `openRequests`/`runningRequests`
- A comment to provide clarification about the field. It is easy to confuse if the key is the ID of the request that the user submitted (which is what thought to it to be, hence 1:n) or the ID of the request sent by Central to Sensors.

With the above in mind, each communication loop, could be a single roundtrip or more, between Central and Sensor should have a unique ID.